tests/activity_api_update_test.go

```diff
@@ -0,0 +1,340 @@
+	s := new(ActivityApiUpdateClientTestSuite)
+	suite.Run(t, s)
+}
+
+func (s *ActivityApiUpdateClientTestSuite) waitForChan(ctx context.Context, ch chan struct{}) {
+	s.T().Helper()
+	select {
+	case <-ch:
+	case <-ctx.Done():
+		s.FailNow("context timeout")
+	}
+}
+
+type (
+	ActivityFunctions func() (string, error)
+	WorkflowFunction  func(context2 workflow.Context) (string, error)
+)
+
+func (s *ActivityApiUpdateClientTestSuite) makeWorkflowFunc(activityFunction ActivityFunctions) WorkflowFunction {
+	return func(ctx workflow.Context) (string, error) {
+
+		var ret string
+		err := workflow.ExecuteActivity(workflow.WithActivityOptions(ctx, workflow.ActivityOptions{
+			ActivityID:             "activity-id",
+			DisableEagerExecution:  true,
+			StartToCloseTimeout:    s.startToCloseTimeout,
+			ScheduleToCloseTimeout: s.scheduleToCloseTimeout,
+			RetryPolicy:            s.activityRetryPolicy,
+		}), activityFunction).Get(ctx, &ret)
+		return "done!", err
+	}
+}
+
+func (s *ActivityApiUpdateClientTestSuite) TestActivityUpdateApi_ChangeRetryInterval() {
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+
+	activityUpdated := make(chan struct{})
+
+	var activityCompleted atomic.Int32
+	activityFunction := func() (string, error) {
+		if activityCompleted.Load() == 0 {
+			activityErr := errors.New("bad-luck-please-retry")
+			activityCompleted.Add(1)
+			return "", activityErr
+		}
+
+		s.waitForChan(ctx, activityUpdated)
+		activityCompleted.Add(1)
+		return "done!", nil
+	}
+
+	workflowFn := s.makeWorkflowFunc(activityFunction)
+
+	s.Worker().RegisterWorkflow(workflowFn)
+	s.Worker().RegisterActivity(activityFunction)
+
+	wfId := "functional-test-activity-update-api"
+	workflowOptions := sdkclient.StartWorkflowOptions{
+		ID:        wfId,
+		TaskQueue: s.TaskQueue(),
+	}
+
+	workflowRun, err := s.SdkClient().ExecuteWorkflow(ctx, workflowOptions, workflowFn)
+	s.NoError(err)
+
+	s.EventuallyWithT(func(t *assert.CollectT) {
+		description, err := s.SdkClient().DescribeWorkflowExecution(ctx, workflowRun.GetID(), workflowRun.GetRunID())
+		assert.NoError(t, err)
+		assert.Equal(t, 1, len(description.PendingActivities))
+		assert.Equal(t, int32(1), activityCompleted.Load())
+	}, 10*time.Second, 500*time.Millisecond)
+
+	updateRequest := &workflowservicepb.UpdateActivityOptionsByIdRequest{
+		Namespace:  s.Namespace(),
+		WorkflowId: workflowRun.GetID(),
+		ActivityId: "activity-id",
+		ActivityOptions: &activitypb.ActivityOptions{
+			RetryPolicy: &commonpb.RetryPolicy{
+				InitialInterval: durationpb.New(1 * time.Second),
+			},
+		},
+		UpdateMask: &fieldmaskpb.FieldMask{Paths: []string{"retry_policy.initial_interval"}},
+	}
+	resp, err := s.FrontendClient().UpdateActivityOptionsById(ctx, updateRequest)
+	s.NoError(err)
+	s.NotNil(resp)
+
+	description, err := s.SdkClient().DescribeWorkflowExecution(ctx, workflowRun.GetID(), workflowRun.GetRunID())
+	s.NotNil(description)
```

Comment:
You can assume that describe response will not be nil and if there's an error, you'll want the test to fail with that error so we know what happened.

```suggestion
	s.NoError(err)
```

Reply:
done